### PR TITLE
tetragon: fix the process exit signal when core dumped

### PR DESCRIPTION
**Pull request by justin0u0**
### Description

The ProcessExit event returns a signal that the process received when it exited. However, core dumped signals (SIGSEGV, SIGILL, etc.) are not returned.

This patch fixes the issue by checking if the process exit code has the core dump bit set (7th bit, 0x80), and returns the correct signal. I used the kernel [code](https://elixir.bootlin.com/linux/v5.19/source/kernel/signal.c#L2076-L2078) as a reference to fix it.

### How to Verify

Launch the `tetra` CLI with the following command:

```bash
tetra getevents --process "run" --output compact
```

Then launch a random program and terminate it using different signals:

```bash
./run &
kill -9 $!
./run &
kill -11 $!
```

**Expected Behavior:**

Before: the signal for core dumped processes is not displayed correctly.

before

After: the correct signal for core dumped processes is displayed.
after

### CODE REVIEW

**Comment by justin0u0** [comment]
> Hey welcome to Tetragon 👋 , thanks for writing this patch!
> 
> Since you are adding something that was missing maybe we could add a test for your change, you can take a look at tests in `pkg/sensors/exec/excit_test.go` for example. We have some programs in `contrib/tester-progs` that you could reuse just to start some process that waits forever I think.
> 
> Please reach out if you need more pointers, that would be a pleasure.

Hi @mtardy,

I've added a test to verify process exit signals from numbers 1 to 15.

Since I couldn’t find a program in contrib/tester-progs that fits my use case, I added a simple pause program to achieve this.

Please take a look when you can. Thanks!

File: contrib/tester-progs/pause
Comment (mtardy) [inline contrib/tester-progs/pause:1]: can you add this to the `contrib/tester-progs/.gitingore` and not commit the binary :)?

**Review by mtardy — COMMENTED** [review]
Thanks that's great :) just a small git related change:

File: contrib/tester-progs/pause
Comment (justin0u0) [reply contrib/tester-progs/pause:1]: Fixed it, thanks for the reminder!

**Pull request merged**